auth0/resource_auth0_action.go

```diff
@@ -251,6 +259,61 @@ func deployAction(ctx context.Context, d *schema.ResourceData, m interface{}) di
 	return diag.FromErr(d.Set("version_id", actionVersion.GetID()))
 }
 
+func preventErasingUnmanagedSecrets(d *schema.ResourceData, api *management.Management) diag.Diagnostics {
+	if !d.HasChange("secrets") {
+		return nil
+	}
+
+	preUpdateAction, err := api.Action.Read(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	oldSecrets, newSecrets := d.GetChange("secrets")
+
+	return checkForUnmanagedActionSecrets(
+		oldSecrets.([]interface{}),
+		newSecrets.([]interface{}),
+		preUpdateAction.Secrets,
+	)
+}
+
+func checkForUnmanagedActionSecrets(
+	oldSecretsFromConfig []interface{},
+	newSecretsFromConfig []interface{},
+	secretsFromAPI []*management.ActionSecret,
+) diag.Diagnostics {
+	secretKeysInConfigMap := make(map[string]bool, len(secretsFromAPI))
+
+	for _, oldSecret := range oldSecretsFromConfig {
+		secretKeyName := oldSecret.(map[string]interface{})["name"].(string)
+		secretKeysInConfigMap[secretKeyName] = true
+	}
+
+	for _, newSecret := range newSecretsFromConfig {
+		secretKeyName := newSecret.(map[string]interface{})["name"].(string)
+		secretKeysInConfigMap[secretKeyName] = true
+	}
+
+	var warnings diag.Diagnostics
```

Comment:
`warnings` is a bit of a misnomer because this collection only will contain errors. I think the convention we've begin to establish is to refer to a collection of `diag.Diagnostics` as `var diags diag.Diagnostics` because they could, in theory, contain both warnings and errors.